public/jira/rolledup/work-type/work-type.ts

```diff
@@ -0,0 +1,182 @@
+import {
+  rollupGroupedHierarchy,
+  groupIssuesByHierarchyLevelOrType,
+  zipRollupDataOntoGroupedData,
+  IssueOrRelease,
+  isDerivedRelease,
+} from "../../rollup/rollup";
+import { getStartAndDueData, mergeStartAndDueData } from "../../rollup/dates/dates";
+import {
+  workType as workTypes,
+  WorkType,
+  DerivedWorkStatus,
+} from "../../derived/work-status/work-status";
+import { RollupLevelAndCalculation } from "../../shared/types";
+import { StartData, DueData } from "../../../shared/issue-data/date-data";
+import { DerivedWorkTiming } from "../../derived/work-timing/work-timing";
+
+// TODO:
+
+// this is more like "derived" from "rollup"
+
+// given some "rolled up" dates ....
+
+// Go to each item ... get it's children ... filter by work status type ...
+// add those as children ...
+
+export type DateAndIssueKeys = StartData &
+  DueData & {
+    issueKeys: string[];
+  };
+
+export type WorkTypeRollups = {
+  [key in WorkType]?: Partial<DateAndIssueKeys>;
+};
+
+export type WithChildren = {
+  children: Partial<DateAndIssueKeys>;
+};
+
+export type WithWorkTypeTiming = {
+  workTypeRollups: WorkTypeRollups & WithChildren;
+};
+
+export type WorkTypeChildRollups = {
+  self: WorkTypeRollups;
+  combined: WorkTypeRollups;
+  children: WorkTypeRollups;
+};
+
+function notEmpty<TValue>(value: TValue | null | undefined): value is TValue {
+  return value !== null && value !== undefined;
+}
+
+export function addWorkTypeDates<T>(
+  issuesOrReleases: IssueOrRelease<T>[],
+  rollupTimingLevelsAndCalculations: RollupLevelAndCalculation[]
+): IssueOrRelease<T & WithWorkTypeTiming>[] {
+  const groupedIssues = groupIssuesByHierarchyLevelOrType(
+    issuesOrReleases,
+    rollupTimingLevelsAndCalculations
+  );
+  // const rollupMethods = rollupTimingLevelsAndCalculations
+  //   .map((rollupData) => rollupData.calculation)
+  //   .reverse();
+  const rolledUpDates = rollupWorkTypeDates(groupedIssues);
+  const zipped = zipRollupDataOntoGroupedData(groupedIssues, rolledUpDates, (item, values) => ({
+    ...item,
+    workTypeRollups: values,
+  }));
+  return zipped.flat();
+}
+
+/**
+ *
+ * @param {Array<import("../rollup").IssuesOrReleases>} issuesOrReleases Starting from low to high
+ * @param {Array<String>} methodNames Starting from low to high
+ * @return {Array<RollupDateData>}
+ */
+export function rollupWorkTypeDates<T>(groupedHierarchy: IssueOrRelease<T>[][]) {
+  return rollupGroupedHierarchy(groupedHierarchy, {
+    createRollupDataFromParentAndChild(issueOrRelease, children: WorkTypeChildRollups[]) {
+      //const methodName = methodNames[hierarchyLevel] || "childrenFirstThenParent";
```

Comment:
So it looks like methodNames is just not part of the function anymore?  This commented out code can be removed then